### ISSUE TITLE
Provide a better way to react to button actions

### ISSUE THREAD

**Issue by nicopap** [nicopap/ui-navigation#22]
Currently, the simplest way to react to a button change is the following:
```rust
fn handle_ui(
  mut events: EventReader<NavEvent>,
  buttons: Query<&MainMenuButton>,
) {
  for event in events.iter() {
    if let NavEvent::NoChanges { from, request: NavRequest::Action } = event {
      match buttons.get(from.first()) {
        // Do things when specific button is activated
        Ok(MainMenuButton::Start) => {}
        Ok(MainMenuButton::Options) => {}
        Ok(MainMenuButton::Exit) => {}
        Err(_) => {}
      }
    }
  }
}
```

This is almost as terrible as declaring the native bevy ui.

There should be a shortcut. Idea is as following:

```rust
fn by_request<T: WorldQuery>(&self, query: Query<T>) -> Option<(NavRequest, QueryItem<Query<T>>)> {}
```

This way it could by called as follow:
```rust
fn handle_ui(
  mut events: EventReader<NavEvent>,
  buttons: Query<&MainMenuButton>,
) {
  for event in events.iter() {
    match event.by_request(query) {
      Some((NavRequest::Action, MainMenuButton::Start)) => {},
      // etc.
```

We could also specify a family of methods, or even an extension trait on `EventReader<NavEvent>` to make it even more convenient. For example specialized a `activated` method that returns a `Option<QueryItem<Query<T>>>` only when the request type is `Action`.

**Comment by nicopap** [comment]
A trait extension could look like this:
```rust
for activated in events.handle(NavRequest::Action, query) {
    match activated {
        MainMenuButton::StartGame => {
            //...
        }
        MainMenuButton::Options => {
            //...
        }
        MainMenuButton::Exit => {
            //...
        }
    }    
}

**Comment by nicopap** [comment]
Maybe even define a custom `WorldQuery` that holds both a `EventReader<NavEvent>` and an arbitrary `T: WorldQuery` and define iteration over that.

Then the API would look like this:
```rust
fn handle_ui(buttons: NavEventQuery<&MainMenuButton>) {
  for button in buttons.iter_activated() {
    match button {
      MainMenuButton::StartGame => {}
      MainMenuButton::Options => {}
      MainMenuButton::Exit => {}
    }
  }
}
```

**Comment by nicopap** [comment]
The `event_helpers` module is intended to fix this issue see 8f0f7626c3daecdf609589558076beb9b6133626
